protected/models/SearchNode.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+/**
+ * Class SearchNode
+ */
+abstract class SearchNode extends CFormModel
```

Comment:
What does extending CFormModel offer us here?

Reply:
I believe it allows the form data to be stored in memory rather than against the database (although admittedly CModel/BaseModel probably operate much the same).